### DISCUSSION TITLE
list_string_agg with custom separator

### DISCUSSION THREAD

**Discussion by Admolly** [duckdb/duckdb#6641] [General]
I don't know if this is possible already or not, but it would be convenient if one could concatenate a list of strings (or column of lists) with a custom separator, similar to the python join function.  I know of the list_string_agg function, but it only allows you to concat with a comma as a separator.

**Comment by Mytherin** [comment]
This is already possible:

```sql
select list_aggr(['a', 'b'], 'string_agg', '|') as l;
┌─────────┐
│    l    │
│ varchar │
├─────────┤
│ a|b     │
└─────────┘
```

**Reply by Admolly** [reply]
ah OK, thanks I figured there was probably a way to do it.
